### PR TITLE
`IN (SELECT ...)` syntax for beam

### PR DESCRIPTION
**Pull request by tathougies**
This is some basic functionality that's been missing for a while.

### CODE REVIEW

File: beam-postgres/test/Database/Beam/Postgres/Test/Select.hs [hunk 1]
```diff
@@ -99,6 +99,14 @@ testInRowValues getConn = testCase "IN with row values works" $
       return $ pair `in_` [pair, pair]
     assertEqual "result" [True] result
 
+testInSelect :: IO ByteString -> TestTree
```
Comment (kmicklas) [inline beam-postgres/test/Database/Beam/Postgres/Test/Select.hs:102]: It looks like this isn't in the test tree?

File: beam-core/beam-core.cabal [hunk 2]
```diff
@@ -2,7 +2,7 @@
 -- see http://haskell.org/cabal/users-guide/
 
 name:                beam-core
-version:             0.9.2.0
+version:             0.9.3.0
```
Comment (kmicklas) [inline beam-core/beam-core.cabal:5]: I believe by the PVP this counts as a breaking change, even though it won't likely affect regular users. I wanted to get as much non-breaking stuff out of the backlog first before getting to breaking changes, but I think we're in a pretty good state now to start merging things for an eventual major release. Can you bump to 0.10?

File: beam-migrate/Database/Beam/Haskell/Syntax.hs [hunk 3]
```diff
@@ -696,6 +696,7 @@ instance IsSql92ExpressionSyntax HsExpr where
   leE q a b = hsApp (hsVar "leE")   [hsMaybe q, a, b]
 
   inE a b = hsApp (hsVar "inE") [a, hsList b]
+  inSelectE _ _ = error "inSelectE"
```
Comment (kmicklas) [inline beam-migrate/Database/Beam/Haskell/Syntax.hs:699]: I was going to ask why the `error` and then noticed there are already many more. :cry:

File: beam-sqlite/beam-sqlite.cabal [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name:                beam-sqlite
```
Comment (kmicklas) [inline beam-sqlite/beam-sqlite.cabal:1]: We now have select tests for SQLite, can you add one here too?

**Review by kmicklas — COMMENTED** [review]
Thanks for doing this! I started working on this in 2019 but never got around to finishing it.

File: beam-core/beam-core.cabal (hunk 2 above)
Comment (tathougies) [reply beam-core/beam-core.cabal]: You're right!

**Pull request merged**